flow/scripts/synth_hier_report.tcl

```diff
@@ -63,18 +58,17 @@ proc write_keep_hierarchy {} {
     set split_cont1 [split $contents1 "\n"]
     foreach line $split_cont1 {
       if {[regexp { +Chip area for top module '(\S+)': (.*)} $line -> module_name area]} {
-        if {[expr $area > $ungroup_threshold]} {
-            puts "Preserving module: $module_name (area: $area)"
-            puts $out_script_ptr "select -module {$module_name}"
-            puts $out_script_ptr "setattr -mod -set keep_hierarchy 1"
-            puts $out_script_ptr "select -clear"
-        } else {
-          puts "Flattening module $module_name (area: $area)"
-        }
+        lappend areas "$area $module_name"
       }
     }
     file delete -force $::env(REPORTS_DIR)/synth_hier_stat_temp_module.txt
   }
+  set areas [lsort -index 0 -real $areas]
+
+  set out_script_ptr [open $::env(SYNTH_STOP_MODULE_SCRIPT) w]
```

Comment:
SYNTH_STOP_MODULE_SCRIPT is no longer a good name as this is just an area list not a stop module script.

Reply:
I view it as a separate concern. I have no opinion about the names, so it will be a dance of dream up name, change, review, change, etc

Reply:
I view a file's name its content as a single concern.  How about:
```
export SYNTH_AREA_PER_MODULE = $(RESULTS_DIR)/synth_area_per_module.txt
```

Reply:
synth_stats.txt?

Reply:
I view the actual statistics as an implementation detail, could be something else in the future.

Reply:
Fine.  I hope this whole thing goes away with @povik's work.